pkgs/development/libraries/science/math/cudnn/extension.nix

```diff
@@ -0,0 +1,112 @@
+    supportedFile = files: findFirst isSupported null files;
+    # Supported versions with versions as keys and file as value
+    supportedVersions = filterAttrs (version: file: file !=null ) (mapAttrs (version: files: supportedFile files) cuDnnVersions);
+    # Compute versioned attribute name to be used in this package set
+    computeName = version: "cudnn_${toUnderscore version}";
+    # Add all supported builds as attributes
+    allBuilds = mapAttrs' (version: file: nameValuePair (computeName version) (buildCuDnnPackage (removeAttrs file ["fileVersion"]))) supportedVersions;
+    # Set the default attributes, e.g. cudnn = cudnn_8_3_1;
+    defaultBuild = { "cudnn" = allBuilds.${computeName cuDnnDefaultVersion}; };
+  in allBuilds // defaultBuild;
+
+  cuDnnVersions = let
+    urlPrefix = "https://developer.download.nvidia.com/compute/redist/cudnn";
+  in {
+    "7.4.2" = [
+      rec {
+        fileVersion = "10.0";
+        fullVersion = "7.4.2.24";
+        sha256 = "18ys0apiz9afid2s6lvy9qbyi8g66aimb2a7ikl1f3dm09mciprf";
+        url = "${urlPrefix}/v${majorMinorPatch fullVersion}/cudnn-10.0-linux-x64-v${fullVersion}.tgz";
+        supportedCudaVersions = [ "10.0" ];
+      }
+    ];
+    "7.6.5" = [
+      rec {
+        fileVersion = "10.0";
+        fullVersion = "7.6.5.32";
+        hash = "sha256-KDVeOV8LK5OsLIO2E2CzW6bNA3fkTni+GXtrYbS0kro=";
+        url = "${urlPrefix}/v${majorMinorPatch fullVersion}/cudnn-${cudatoolkit.majorVersion}-linux-x64-v${fullVersion}.tgz";
+        supportedCudaVersions = [ "10.0" ];
+      }
+      rec {
+        fileVersion = "10.1";
+        fullVersion = "7.6.5.32";
+        hash = "sha256-YAJn8squ0v1Y6yFLpmnY6jXzlqfRm5SCLms2+fcIjCA=";
+        url = "${urlPrefix}/v${majorMinorPatch fullVersion}/cudnn-${cudatoolkit.majorVersion}-linux-x64-v${fullVersion}.tgz";
+        supportedCudaVersions = [ "10.1" ];
+      }
+      rec {
+        fileVersion = "10.2";
+        fullVersion = "7.6.5.32";
+        hash = "sha256-fq7IA5osMKsLx1jTA1iHZ2k972v0myJIWiwAvy4TbLN=";
+        url = "${urlPrefix}/v${majorMinorPatch fullVersion}/cudnn-${cudatoolkit.majorVersion}-linux-x64-v${fullVersion}.tgz";
+        supportedCudaVersions = [ "10.2" ];
+      }
+    ];
+    "8.1.1" = [
+      rec {
+        fileVersion = "10.2";
+        fullVersion = "8.1.1.33";
+        hash = "sha256-Kkp7mabpv6aQ6xm7QeSVU/KnpJGls6v8rpAOFmxbbr0=";
+        url = "${urlPrefix}/v${majorMinorPatch fullVersion}/cudnn-${fileVersion}-linux-x64-v${fullVersion}.tgz";
+        supportedCudaVersions = [ "10.2" ];
+      }
+      rec {
+        fileVersion = "11.2";
+        fullVersion = "8.1.1.33";
+        hash = "sha256-mKh4TpKGLyABjSDCgbMNSgzZUfk2lPZDPM9K6cUCumo=";
+        url = "${urlPrefix}/v${majorMinorPatch fullVersion}/cudnn-${fileVersion}-linux-x64-v${fullVersion}.tgz";
+        supportedCudaVersions = [ "11.2" ];
+      }
+    ];
+    "8.3.2" = [
```

Comment:
IIUC there's now a v8.3.3 which also supports CUDA 11.6. I don't think adding it ought to be a blocker on this PR tho.